### PR TITLE
Add union type to ParamBuilder and BuilderHelpers

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: lib/PhpParser/Builder/Param.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function setDefault($value) {
     /**
      * Sets type for the parameter.
      *
-     * @param string|Node\Name|Node\NullableType $type Parameter type
+     * @param string|Node\Name|Node\NullableType|Node\UnionType $type Parameter type
```
Comment (nikic) [inline lib/PhpParser/Builder/Param.php]: normalizeType() needs to be changed to allow it as well.

**Comment by TomasVotruba** [comment]
Just randomly came here... is anything missing to merge?

File: lib/PhpParser/BuilderHelpers.php [hunk 2]
```diff
@@ -158,9 +158,9 @@ private static function normalizeNameCommon($name, bool $allowExpr) {
      * In particular, builtin types become Identifiers, custom types become Names and nullables
      * are wrapped in NullableType nodes.
      *
-     * @param string|Name|Identifier|NullableType $type The type to normalize
+     * @param string|Name|Identifier|NullableType|UnionType $type The type to normalize
```
Comment (nikic) [inline lib/PhpParser/BuilderHelpers.php]: Missing use.

File: lib/PhpParser/BuilderHelpers.php [hunk 3]
```diff
@@ -158,9 +158,9 @@ private static function normalizeNameCommon($name, bool $allowExpr) {
...
      *
-     * @return Name|Identifier|NullableType The normalized type
+     * @return Name|Identifier|NullableType|UnionType The normalized type
      */
     public static function normalizeType($type) {
         if (!is_string($type)) {
```
Comment (nikic) [inline lib/PhpParser/BuilderHelpers.php]: UnionType needs to be allowed in the check below.

**Comment by TomasVotruba** [comment]
I'll look on it

**Comment by TomasVotruba** [comment]
> This needs a test.

Where would you test it? 
There is some test:
https://github.com/nikic/PHP-Parser/blob/master/test/code/parser/stmt/function/unionTypes.test

File: test/PhpParser/Builder/ParamTest.php [hunk 4]
```diff
@@ -114,6 +116,12 @@ public function provideTestTypes() {
...
             ['self', new Node\Name('self')],
+            [new Node\Name('Some\Class'), new Node\Name('Some\Class')],
+        ];
+    }
+
+    public function provideTestNullableTypes() {
+        return [
             ['?array', new Node\NullableType(new Node\Identifier('array'))],
             ['?Some\Class', new Node\NullableType(new Node\Name('Some\Class'))],
             [new Node\Name('Some\Class'), new Node\Name('Some\Class')],
```
Comment (nikic) [inline test/PhpParser/Builder/ParamTest.php]: This line is duplicated from above.

Comment (TomasVotruba) [reply test/PhpParser/Builder/ParamTest.php]: Fixed

**Comment by nikic** [comment]
Thanks!

**Comment by TomasVotruba** [comment]
You're welcome :+1:

**Pull request merged**